coalib/parsing/InvalidFilterException.py

```diff
@@ -1,2 +1,9 @@
 class InvalidFilterException(Exception):
-    pass
+    def __init__(self, filter_name=None, available_filters=None):
+        self.filter_name = filter_name
+        self.available_filters = sorted(available_filters)
+        message = {self.filter_name +
+                   ' is an invalid filter. Available filters: ' +
+                   ', '.join(self.available_filters)}
```

Comment:
Using too much `+` to build strings is considered bad style. It makes it also more complicated to properly care for whitespace. Look at this:

```python
'{!r} is an invalid filter. Available filters: {}'
.format(filter_name, ', '.join(self.available_filters))
```

It will also automatically use `repr(format_string)` because of `{!r}`, putting quotes around it for better readability, which takes me to the next point :) Why `', '.join(...)` at all. Just pass `self.available_filters` to `.format(...)`. It will give you normal list syntax output like `['filter0', 'filter1', ...]`. Also not bad... ;) And will also nicely handle the case of no available filters by just outputting `[]`. Your choice :)